### PR TITLE
fix(Script/ICC): Spirit Alarm and Deathbound Ward

### PR DESCRIPTION
**Pull request by HoustonCore**
## CHANGES PROPOSED:
- Start to move waypoints event after activate traps
- Adding waypoints for Deathbound Ward
- Source https://github.com/WarheadCore/WarheadCore/commit/3d1f68f40dffbef0e2da62c5ec3dc923c41d6e4e

## TESTS PERFORMED:
- Build and test in-game
- Need help with the correct selection of Entry waypoints

## HOW TO TEST THE CHANGES:
- Go to ICC
- Activate trasp and see

## Target branch(es):
- [x] Master

 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Comment by Viste** [comment]
Где снифы :DD

File: src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp [hunk 1]
```diff
@@ -1949,34 +1949,38 @@ class spell_icc_sprit_alarm : public SpellScriptLoader
...
                     case EVENT_AWAKEN_WARD_4:
-                        range = 50.0f;
+                        trapId = GO_SPIRIT_ALARM_4;
                         break;
                     default:
                         return;
                 }
 
+                if (GameObject* trap = GetCaster()->FindNearestGameObject(trapId, 5.0f))
+                    trap->SetRespawnTime(trap->GetGOInfo()->GetAutoCloseTime() / IN_MILLISECONDS);
```
Comment (FrancescoBorzi) [inline src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp:1972]: please add brackets

File: src/server/scripts/Northrend/IcecrownCitadel/instance_icecrown_citadel.cpp [hunk 2]
```diff
@@ -1861,6 +1861,31 @@ class instance_icecrown_citadel : public InstanceMapScript
...
+                    trapPositions = GoSpiritAlarm_2;
+                    break;
+                case GO_SPIRIT_ALARM_3:
+                    trapPositions = GoSpiritAlarm_3;
+                    break;
+                case GO_SPIRIT_ALARM_4:
+                    trapPositions = GoSpiritAlarm_4;
+                    break;
+                default:
+                    return;
```
Comment (FrancescoBorzi) [inline src/server/scripts/Northrend/IcecrownCitadel/instance_icecrown_citadel.cpp:1883]: indentation

**Comment by HoustonCore** [comment]
> its from sniffs? why you  :D (its just local meme don't count this) 

No sniffs, i made it all up myself)

**Comment by HoustonCore** [comment]
> its from sniffs? why you  :D (its just local meme don't count this) 

No sniffs, i made it all up myself)

**Comment by Krath** [comment]
Tested it and it looks good. The deathbound ward start patrolling when you step on their traps

File: src/server/scripts/Northrend/IcecrownCitadel/instance_icecrown_citadel.cpp (hunk 2 above)
Comment (HoustonCore) [reply src/server/scripts/Northrend/IcecrownCitadel/instance_icecrown_citadel.cpp:1883]: Where?)

File: src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp (hunk 1 above)
Comment (HoustonCore) [reply src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp:1973]: Done

File: src/server/scripts/Northrend/IcecrownCitadel/instance_icecrown_citadel.cpp (hunk 2 above)
Comment (Viste) [reply src/server/scripts/Northrend/IcecrownCitadel/instance_icecrown_citadel.cpp:1883]: case через 4 пробела от {

Comment (HoustonCore) [reply src/server/scripts/Northrend/IcecrownCitadel/instance_icecrown_citadel.cpp:1883]: Ops.. I saw, fix later.

Comment (HoustonCore) [reply src/server/scripts/Northrend/IcecrownCitadel/instance_icecrown_citadel.cpp]: Done

**Review by FrancescoBorzi — APPROVED** [review]

**Comment by FrancescoBorzi** [comment]
ready to be merged! good job @HoustonCore, thanks @Krath for testing!

**Comment by FrancescoBorzi** [comment]
@HoustonCore can you please fix the merge conflicts? so we can merge this

**Comment by HoustonCore** [comment]
@FrancescoBorzi I don't support development anymore. If you want, you can fix it yourself.

**Pull request closed without merging**